src/stc/scintilla/lexers/LexMMIXAL.cxx

```diff
@@ -103,9 +103,7 @@ static void ColouriseMMIXALDoc(Sci_PositionU startPos, Sci_Position length, int
 				char s[100];
 				sc.GetCurrent(s, sizeof(s));
 				if (*s == ':') {	// ignore base prefix for match
-					for (size_t i = 0; i != sizeof(s); ++i) {
-						*(s+i) = *(s+i+1);
-					}
+					memmove(s + 1, s, sizeof(s) - 1);
```

Comment:
This code isn't equivalent... The original version shifts `s` one character to the left, while the new one shifts it to the right.

AFAICS the only thing which needs to be fixed here is to use `sizeof(s) - 1` in the loop.

Reply:
you are right, I mixed up source and destination. I will fix it

Reply:
Done, thank you for review. _memmove_ is faster than raw copying loop because of using big registers on copy (xmm, and GPR).

Reply:
GCC optimizer implicitly replaced raw loop by memmove - performance is main reason